app/services/prison_visits/client.rb

```diff
@@ -52,8 +55,25 @@ def request(method, route, params)
       response = @connection.request(options)
 
       JSON.parse(response.body)
+    rescue Excon::Errors::HTTPStatusError => e
```

Comment:
My suggestions to clean up this large complex method are:
- break up the method in 2, with one of them having the same body as before this PR, and the other one handling the errors
- rescue the top level Excon error (`Excon::Errors::Error`) only and construct the new APIError error message with the rescued class name and and other info
- Not sure with the value of adding more information than the error type and the rescued error message really, since we should be able to correlate the request that failed in Kibana easily. I'd added later more detail if we find that it would be useful.
- If you end up keeping this code consider changing `e.class` for `e.inspect` on line 73 so we the error message.